pymeasure/instruments/instrument.py

```diff
@@ -136,16 +187,29 @@ def control(get_command, set_command, docs,
                             before value mapping, returning the processed value
         :param set_process: A function that takes a value and allows processing
                             before value mapping, returning the processed value
+        :param command_process: A function that take a command and allows processing
+                            before executing the command, for getting
         :param check_set_errors: Toggles checking errors after setting
         :param check_get_errors: Toggles checking errors after getting
+        :param no_getting: Disable fget
+        :param no_setting: Disable fset
+        :param dynamic: boolean to indicate whether property is dynamic
         """
 
-        if map_values and isinstance(values, dict):
-            # Prepare the inverse values for performance
-            inverse = {v: k for k, v in values.items()}
-
-        def fget(self):
-            vals = self.values(get_command, **kwargs)
+        if dynamic:
```

Comment:
Well, in this case I have used the function `vars` to retrieve all the parameters of the function, to avoid to mention them one by one and be sure that if a parameter is added or removed this will be taken into account. To do so, the `vars` function must be called at the beginning of function to avoid that it will report other local variables possibly used in the implementation. Another possibility is to compute the `dyn_list` anyway at the beginning of function and avoid the `if` statement. Any idea suggestion from your side ?

Reply:
New changes implemented your suggestion.